### PR TITLE
Added contributors from the hole2 analysis module

### PR DESCRIPTION
**Pull request by ianmkenney**
- Added all authors and co-authors from:
  - `MDAnalysis/analysis/hole2/hole.py`
  - `MDAnalysis/analysis/hole2/__init__.py`
  - `MDAnalysis/analysis/hole2/templates.py`
  - `MDAnalysis/analysis/hole2/utils.py`
- Co-author order was determined from commit messages

### CODE REVIEW

**Comment by ianmkenney** [comment]
@IAlibay mind taking a look?

File: AUTHORS.md [hunk 1]
```diff
@@ -18,8 +18,26 @@ The rules for this file:
...
   * Don't ever delete anything
 -->
+**2020**
+- Lily Wang \<@lilyminium\>
+- Irfan Alibay \<@IAlibay\>
+- Oliver Beckstein \<@orbeckst\>
+- Philip Loche \<@PicoCentauri\>
+
+**2021**
+- Lily Wang \<@lilyminium\>
```
Comment (IAlibay) [inline AUTHORS.md]: I probably wouldn't double up on each year? (at least we don't in core, I'm not sure if there's a different standard for this)

**Review by IAlibay — CHANGES_REQUESTED** [review]
So I think this might be a bit more complicated than this unfortunately. The hole2 code was derived from the original hole code (see https://github.com/MDAnalysis/mdanalysis/blob/release-1.1.1/package/MDAnalysis/analysis/hole.py), making it a derivative. 

@lilyminium and @orbeckst might be better able to weigh in on how derivative it is.

That being said you'll probably want to add the contributors from that file too.

**Comment by orbeckst** [comment]
The safe approach is to include everyone from the earlier version. It wouldn't be fair to say that the hole2 rewrite is independent from the earlier code.

If we want to relicense and someone is not responding then we can dig in and assess on a case-by-case basis if the contribution is likely copyright worthy.

File: AUTHORS.md (hunk 1 above)
Comment (orbeckst) [reply AUTHORS.md]: Just do first year.

I also wouldn't backslash-escape: the file is primarily for humans.

**Review by orbeckst — CHANGES_REQUESTED** [review]
As @IAlibay said: include the authors from the previous code, please.

EDIT: Also update the README to point to AUTHORS for the copyright line.

File: AUTHORS.md (hunk 1 above)
Comment (ianmkenney) [reply AUTHORS.md]: Removed duplicates and removed backslashes

**Review by orbeckst — APPROVED** [review]

**Review by IAlibay — APPROVED** [review]
lgtm!

**Pull request merged**